include/s3select.h

```diff
@@ -1087,20 +1087,20 @@ void push_data_type::operator()(s3select* self, const char* a, const char* b) co
 
   if(cast_operator("int"))
   {
-    self->getAction()->dataTypeQ.push_back("int");
+    self->getAction()->dataTypeQ.emplace_back("int");
```

Comment:
ok, it performs better.
should note, those functions (AST builder) used for building the AST and are not intensive.